src/Microsoft.DotNet.ProjectModel/ProjectPath.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.DotNet.ProjectModel
+{
+    internal static class ProjectPath
+    {
+        public static string NormalizeProjectDirectoryPath(string path)
+        {
+            string fullPath = Path.GetFullPath(path);
+
+            if (IsProjectFilePath(fullPath))
+            {
+                return Path.GetDirectoryName(fullPath);
+            }
+            else if (IsDirectoryContainingProjectFile(fullPath))
+            {
+                return fullPath;
+            }
+
+            return null;
```

Comment:
I was about to say: "returns `null`? how reckless" and then I found out we did that before. "well, OK then..."